### PR TITLE
Expression equality checking, some optimizations

### PR DESCRIPTION
**Pull request by PThorpe92**
This PR mainly adds custom logic to check equality in ast expressions. Not sure if this belongs in the `vendored` parser or not, let me know and I'll bring it out. Also replaces `Vec` arguments with slice refs where possible, as well as some clippy warnings in the same `emitter` file.

I'll write some more tests tomorrow to make sure this is as thorough as possible.

EDIT: failed test same issue referenced in #484. Marking as draft until more tests + cases added

### CODE REVIEW

**Comment by PThorpe92** [comment]
I'm sure I'm missing some opportunity to optimize this.  

Also, thoughts on moving this out of `vendored` ?

**Comment by jussisaurio** [comment]
> I'm sure I'm missing some opportunity to optimize this.
> 
> Also, thoughts on moving this out of `vendored` ?

For now I would keep the surface area of the things we change in the vendored parser as low as possible, so IMO migrate this elsewhere 👍

File: vendored/sqlite3-parser/src/parser/ast/mod.rs [hunk 1]
```diff
@@ -534,6 +679,27 @@ impl Literal {
...
+    fn is_equivalent(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Numeric(n1), Self::Numeric(n2)) => {
+                match (normalize_numeric_str(n1), normalize_numeric_str(n2)) {
+                    (Some(canonical), Some(canonical2)) => canonical == canonical2,
+                    _ => false,
+                }
+            }
+            // TODO: check for quoted == unquoted strings?
+            (Self::String(s1), Self::String(s2)) => s1.eq_ignore_ascii_case(s2),
```
Comment (jussisaurio) [inline vendored/sqlite3-parser/src/parser/ast/mod.rs:693]: check this PR out: https://github.com/tursodatabase/limbo/pull/492

would be great to handle the same cases for equivalency, although ideally without extra allocations

Comment (jussisaurio) [reply vendored/sqlite3-parser/src/parser/ast/mod.rs:693]: actually that PR was only for identifiers... so we could use the same logic for Id and Qualified

Comment (PThorpe92) [reply vendored/sqlite3-parser/src/parser/ast/mod.rs]: All set :+1:

**Comment by PThorpe92** [comment]
Alright I think that is all the cases that make sense at this point. I didn't go as far as to try to eval and compute any possible `constexpr` values.

I noticed the help msg is getting much too long so I trimmed it down a bit. @penberg just fyi I saw your comment on the last PR about the `sqlite3` shell tests/compatibility :+1: it's noted and I'll have to add some logic that skips all the `limbo` specific output. 

If 1:1 identical output to sqlite is the goal (minus the prompt/version/etc), I can keep that in mind for future shell/UI features.

**Pull request merged**